worker/src/sender/mod.rs

```diff
@@ -0,0 +1,39 @@
+use std::{net::SocketAddr, sync::Arc};
+
+use proto::common::{
+    instance::{self, InstanceId},
+    node::Metrics,
+};
+use reqwest::Client;
+
+/// Sends a http request to `ctl::http` component with body
+/// containing `instance::Status` information used in `ctl::discovery`
+#[derive(Clone, Debug)]
+pub struct Sender {
+    pub client: Client,
+    pub http_path: SocketAddr,
+}
+
+impl Sender {
+    /// Creates a `Sender` instance wrapped by an `Arc`.
+    #[must_use]
+    pub fn new(path: SocketAddr) -> Arc<Self> {
```

Comment:
In general, `new` methods return `Self`. This implementation forces downstream users to use an `Arc`, which is not always desirable